Sentry.php

```diff
@@ -0,0 +1,134 @@
+ * Free Software Foundation with the addition of the following permission added
+ * to Section 15 as permitted in Section 7(a): FOR ANY PART OF THE COVERED WORK
+ * IN WHICH THE COPYRIGHT IS OWNED BY SUGARCRM, SUGARCRM DISCLAIMS THE WARRANTY
+ * OF NON INFRINGEMENT OF THIRD PARTY RIGHTS.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program; if not, see http://www.gnu.org/licenses or write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+class Sentry
+{
+    protected static $enabled;
+    protected static $dsn;
+    protected static $instance;
+    protected static $client;
+
+    public static function getInstance()
+    {
+        if (self::$instance !== null) {
+            return self::$instance;
+        }
+
+        return self::$instance = new self();
+    }
+
+    public function __construct()
+    {
+        global $sugar_config;
+
+        try {
+            if (!isset($sugar_config['sentry']['dsn'])) {
+                throw new \SuiteCRM\Exception\Exception('You must configure Sentry DSN.');
+            }
+        } catch (\SuiteCRM\Exception\Exception$exception) {
+            LoggerManager::getLogger()->fatal($exception->getMessage() . "\nTrace:\n" . $exception->getTraceAsString());
+        }
+
+        self::$enabled = $sugar_config['sentry']['enabled'];
+        self::$dsn = $sugar_config['sentry']['dsn'];
+        self::$client = $this->buildClient();
+        $this->init();
+    }
+
+    protected function buildClient()
+    {
+        return new Raven_Client(self::$dsn);
+    }
+
+    protected function init()
+    {
+        global $current_user;
+
+        if (self::$enabled) {
+            $user_context = $current_user->id;
+            self::$client->user_context($user_context);
```

Comment:
If there's no user logged in, what does `$current_user->id` return? Just `null`?

Reply:
Can we provide the user's name and/or email as well? Just having the ID doesn't tell us that much.

Reply:
And also the PHP Version and SuiteCRM version, if possible 😇

Reply:
Okay, yeah this is a problem because `$current_user->id` can return an error if `$current_user` is null.

```
ERROR: Trying to get property 'id' of non-object 
in /Users/connorshea/code/sites/suitecrm-upstream/Sentry.php:85
```

Reply:
I think we should actually be handling this in the `handleException` and `captureMessage` functions, rather than in the install step, since it seems like the install hook doesn't actually have a current_user.